src/Hbar.js

```diff
@@ -65,7 +65,13 @@ export default class Hbar {
      * @returns {Hbar}
      */
     static fromString(str, unit = HbarUnit.Hbar) {
-        return new Hbar(new BigNumber(str), unit);
+        const pattern = /^((?:\+|-)?\d+(?:\.\d+)?)(?: (tℏ|μℏ|mℏ|ℏ|kℏ|Mℏ|Gℏ))?$/;
+        if (pattern.test(str)) {
+            let [amount, symbol] = str.split(" ");
+            return new Hbar(new BigNumber(amount), HbarUnit.fromString(symbol));
+        } else {
+            throw new Error("Variable failed regex: "+str);
```

Comment:
For the error just make it a string constant; something like `invalid string provided` or `invalid argument` or something else if you think of one better.